main.go

```diff
@@ -84,6 +105,35 @@ func main() {
 	}
 	apiClient := v1.NewAPI(c)
 
+	if metadataC, err = newClient(metadataC, options...); err != nil {
+		klog.Fatalf("error building custom metadata API client:", err)
+	}
+	apiMetadataClient := v1.NewAPI(metadataC)
+
+	go func() {
+		var err error
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			klog.Infof("refreshing metric metadata")
+			mutex.Lock()
+			metadataCtx, metadataCancel := context.WithTimeout(ctx, 5*time.Second)
+			metricMetadata, err = apiMetadataClient.Metadata(metadataCtx, "", "")
+			mutex.Unlock()
+			if err != nil {
+				klog.Errorf("error refreshing metric metadata: %s", err.Error())
+				time.Sleep(250 * time.Millisecond)
+			} else {
+				time.Sleep(1 * time.Minute)
```

Comment:
This seems like a good candidate for using `time.Ticker`.

Reply:
fixed in https://github.com/authzed/thanos-federate-proxy/pull/1/commits/46d82bf651e58acaf37ac15a728a60b54a6c971a